src/datasets/builder.py

```diff
@@ -882,18 +888,16 @@ class GeneratorBasedBuilder(DatasetBuilder):
     (`_split_generators`). See the method docstrings for details.
     """
 
-    # GeneratorBasedBulder should have dummy data for tests by default
+    # GeneratorBasedBuilder should have dummy data for tests by default
     test_dummy_data = True
 
-    # Batch size used by the ArrowWriter
-    # It defines the number of samples that are kept in memory before writing them
-    # and also the length of the arrow chunks
-    # None means that the ArrowWriter will use its default value
-    _writer_batch_size = None
-
-    def __init__(self, *args, **kwargs):
-        self._writer_batch_size = kwargs.pop("writer_batch_size", self._writer_batch_size)
+    def __init__(self, *args, writer_batch_size=None, **kwargs):
         super(GeneratorBasedBuilder, self).__init__(*args, **kwargs)
+        # Batch size used by the ArrowWriter
+        # It defines the number of samples that are kept in memory before writing them
+        # and also the length of the arrow chunks
+        # None means that the ArrowWriter will use its default value
+        self._writer_batch_size = writer_batch_size
```

Comment:
Sure! I agree it will be more useful with a default class attribute value. :)

Reply:
@lhoestq, BTW, is there any plan to migrate tests from unittest to pytest?

Reply:
We're already using pytest (even though some parts of the test code uses unittest). In what part of the tests would you like to see such changes ?

Reply:
I just mentioned it because the test for this part of the code uses unittest, e.g.

Reply:
What would be the necessary changes ? I guess it's about not using the TestCase class and use plain test functions with fixtures instead. Or are you thinking of other changes ? Also I'm curious about the advantages of such a migration

Reply:
I see !
Moreover since we're starting to have several cases where we have to test many cases through the same code this is something we should put forward in terms of best practice, or we'll end up with the issues that come from code duplication in the test code.
I agree for having that as a background task for now. Thanks for the suggestion !